codalab/worker/file_util.py

```diff
@@ -10,10 +10,15 @@
 
 BINARY_PLACEHOLDER = '<binary>'
 NONE_PLACEHOLDER = '<none>'
+GIT_PATTERN = '.git'
 
 
 def tar_gzip_directory(
-    directory_path, follow_symlinks=False, exclude_patterns=[], exclude_names=[]
+    directory_path,
+    follow_symlinks=False,
+    exclude_patterns=[],
+    exclude_names=[],
+    ignore_file='.gitignore',
```

Comment:
Perhaps this default should be `None`.  In general, should have as little magic/processing as possible.